Aztec/Classes/Public/TextKit/TextStorage.swift

```diff
@@ -111,7 +115,111 @@ public class TextStorage: NSTextStorage {
         edited(.EditedAttributes, range: range, changeInLength: 0)
     }
 
-    // MARK: - Styles
+    // MARK: - Styles: Synchronization with DOM
+
+    /// Copies all styles in the specified range to the DOM.
+    ///
+    /// - Parameters:
+    ///     - range: the range from which to take the styles to copy.
+    ///
+    private func copyStylesToDOM(spanning range: NSRange) {
+
+        let options = NSAttributedStringEnumerationOptions(rawValue: 0)
+
+        textStore.enumerateAttributesInRange(range, options: options) { (attributes, range, stop) in
+            // Edit attributes
+
+            for (key, value) in attributes {
+                switch (key) {
+                case NSFontAttributeName:
+                    copyFontAttributesToDOM(spanning: range, attributeValue: value)
+                case NSStrikethroughStyleAttributeName:
+                    copyStrikethroughStyleToDOM(spanning: range, attributeValue: value)
+                case NSUnderlineStyleAttributeName:
+                    copyUnderlineStyleToDOM(spanning: range, attributeValue: value)
+                default:
+                    break
+                }
+            }
+        }
+    }
+    
+    private func copyFontAttributesToDOM(spanning range: NSRange, attributeValue value: AnyObject) {
+        
+        guard let font = value as? UIFont else {
+            assertionFailure("Was expecting a UIFont object as the value for the font attribute.")
+            return
+        }
+        
+        copyFontAttributesToDOM(spanning: range, font: font)
+    }
+    
+    private func copyFontAttributesToDOM(spanning range: NSRange, font: UIFont) {
+        
+        let fontTraits = font.fontDescriptor().symbolicTraits
+        
+        if fontTraits.contains(.TraitBold) {
+            enableBoldInDOM(range)
+        }
+        
+        if fontTraits.contains(.TraitItalic) {
+            enableItalicInDOM(range)
+        }
+    }
+    
+    private func copyStrikethroughStyleToDOM(spanning range: NSRange, attributeValue value: AnyObject) {
+        
+        guard let intValue = value as? Int else {
+            assertionFailure("The strikethrough style is always expected to be an Int.")
+            return
+        }
+        
+        guard let style = NSUnderlineStyle(rawValue: intValue) else {
+            assertionFailure("The strikethrough style value is not-known.")
+            return
+        }
+        
+        copyStrikethroughStyleToDOM(spanning: range, style: style)
+    }
+    
+    private func copyStrikethroughStyleToDOM(spanning range: NSRange, style: NSUnderlineStyle) {
```

Comment:
It seems counter-intuitive but it's indeed right.  From the documentation of `NSUnderlineStyle`:

> // This defines currently supported values for NSUnderlineStyleAttributeName and NSStrikethroughStyleAttributeName. NSUnderlineStyle*, NSUnderlinePattern*, and NSUnderlineByWord are or'ed together to produce an underline style.